### PR TITLE
`tensorflow`: add `tf.strings` module

### PR DESCRIPTION
**Pull request by hoel-bagard**
Some of the stubs are taken from [here](https://github.com/hmc-cs-mdrissi/tensorflow_stubs/blob/main/stubs/tensorflow/strings.pyi).

@hmc-cs-mdrissi If you have time to check.

### CODE REVIEW

File: stubs/tensorflow/tensorflow/strings.pyi [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+def format(
+    template: str, inputs: TensorCompatible, placeholder: str = "{}", summarize: int = 3, name: str | None = None
+) -> Tensor: ...
+def join(inputs: Sequence[TensorCompatible | RaggedTensor], separator: str = "", name: str | None = None) -> Tensor: ...
+@overload
+def length(input: TensorCompatible, unit: Literal["BYTE", "UTF8_CHAR"] = "BYTE", name: str | None = None) -> Tensor: ...
+@overload
+def length(input: RaggedTensor, unit: Literal["BYTE", "UTF8_CHAR"] = "BYTE", name: str | None = None) -> RaggedTensor: ...
+@overload
+def lower(input: TensorCompatible, encoding: Literal["utf-8", "", " "] = "", name: str | None = None) -> Tensor: ...
```
Comment (JelleZijlstra) [inline stubs/tensorflow/tensorflow/strings.pyi:40]: The docstring says `Allowed values are '' and 'utf-8'.` Why do you also accept `" "`?

File: stubs/tensorflow/tensorflow/strings.pyi [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+) -> RaggedTensor: ...
+def unsorted_segment_join(
+    inputs: StringTensorCompatible,
+    segment_ids: UIntTensorCompatible,
+    num_segments: UIntTensorCompatible,
+    separator: str = "",
+    name: str | None = None,
+) -> Tensor: ...
+@overload
+def upper(input: TensorCompatible, encoding: Literal["utf-8", "", " "] = "", name: str | None = None) -> Tensor: ...
```
Comment (JelleZijlstra) [inline stubs/tensorflow/tensorflow/strings.pyi:239]: Again the docstring doesn't say `" "` is allowed.

File: stubs/tensorflow/tensorflow/strings.pyi (hunk 1 above)
Comment (hoel-bagard) [reply stubs/tensorflow/tensorflow/strings.pyi]: Must have misread the docs, sorry. Fixed in f33b7275b.

File: stubs/tensorflow/tensorflow/strings.pyi (hunk 2 above)
Comment (hoel-bagard) [reply stubs/tensorflow/tensorflow/strings.pyi]: Fixed in ae35a8b3e.

**Pull request merged**